File: s2container.net-example/S2FormExample/S2FormExample.Tests/TestGender.cs
#region Copyright

/*
 * Copyright 2005-2007 the Seasar Foundation and the Others.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */

#endregion

using System.Collections.Generic;
using System.IO;
using System.Reflection;
using log4net;
using log4net.Config;
using log4net.Util;
using MbUnit.Framework;
using Seasar.Extension.Unit;
using Seasar.S2FormExample.Logics.Dao;
using Seasar.S2FormExample.Logics.Dto;

namespace Seasar.S2FormExample.Tests
{
    /// <summary>
    /// 性別用テストケースクラス
    /// </summary>
    [TestFixture]
    public class TestGender : S2TestCase
    {
        /// <summary>
        /// Logic設定ファイル
        /// </summary>
        private const string PATH = "ExampleLogics.dicon";

        /// <summary>
        /// テストのセットアップ
        /// </summary>
        [SetUp]
        public void Setup()
        {
            FileInfo info = new FileInfo(
                string.Format("{0}.dll.config", SystemInfo.AssemblyShortName(
                                                    Assembly.GetExecutingAssembly())));
            // アセンブリがdllの場合は".dll.config"

            XmlConfigurator.Configure(LogManager.GetRepository(), info);
        }

        /// <summary>
        /// DAOのテスト
        /// </summary>
        [Test, S2]
        public void TestDao()
        {
            Include(PATH);

            IGenderDao dao = (IGenderDao) GetComponent(typeof (IGenderDao));

            IList<GenderDto> list = dao.GetAll();
            Assert.AreEqual(2, list.Count, "Count");
        }
    }
}

File: trunk/s2container.net-example/VbS2FormSample/S2FormExample.Tests/TestGender.vb
''
'' Copyright 2005-2008 the Seasar Foundation and the Others.
''
'' Licensed under the Apache License, Version 2.0 (the "License");
'' you may not use this file except in compliance with the License.
'' You may obtain a copy of the License at
''
''     http://www.apache.org/licenses/LICENSE-2.0
''
'' Unless required by applicable law or agreed to in writing, software
'' distributed under the License is distributed on an "AS IS" BASIS,
'' WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
'' either express or implied. See the License for the specific language
'' governing permissions and limitations under the License.
''
''
Imports System.IO
Imports System.Reflection
Imports Seasar.S2FormExample.Logics.Dto
Imports log4net.Config
Imports log4net
Imports Seasar.S2FormExample.Logics.Dao
Imports MbUnit.Framework
Imports Seasar.Quill.Unit
Imports Seasar.Extension.Unit
Imports log4net.Util

''' <summary>
'''  性別用テストケースクラス
''' </summary>
''' <remarks></remarks>
<TestFixture()> _
Public Class TestGender
    Inherits QuillTestCase

    Protected dao As IGenderDao

    ''' <summary>
    ''' テストのセットアップ
    ''' </summary>
    ''' <remarks></remarks>
    <SetUp()> _
    Public Sub Setup()
        Dim info As FileInfo = _
                New FileInfo(Format(SystemInfo.AssemblyShortName(Assembly.GetExecutingAssembly()), _
                                      "{0}.dll.config"))
        XmlConfigurator.Configure(LogManager.GetRepository(), info)
    End Sub

    ''' <summary>
    ''' DAOのテスト
    ''' </summary>
    ''' <remarks></remarks>
    <Test(), Quill(Tx.Rollback)> _
    Public Sub TestDao()

        Dim list As IList(Of GenderDto) = dao.GetAll()
        Assert.AreEqual(2, list.Count, "Count")
    End Sub
End Class